### PR TITLE
Add gdal package to specs-extended 

### PR DESCRIPTION
**Pull request by arc9693**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?

- Add gdal package to specs-extended to fulfil a customer requirement.
- Package gdal depends on following packages which were added as part of this effort:
    1. CharLS
    2. cfitsio
    3. ogdi
    4. libgta
    5. xerces-c
    6. liblerc
    7. shapelib
    8. qt5-serial
    9. blosc
    10. uriparser
    11. freexl
    12. libgeotiff
    13. proj
    14. gpsbabel
    15. hdf
    16. netcdf
    17. libkml

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Pipeline build id: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=424070&view=results

### CODE REVIEW

File: SPECS-EXTENDED/gdal/gdal.spec [hunk 1]
```diff
@@ -0,0 +1,1255 @@
...
+Source3:        gdal-config
+# Add some utils to the default install target
+Patch0:         gdal_utils.patch
+%global run_tests 1
+%global bashcompletiondir %(pkg-config --variable=compatdir bash-completion)
+# We have multilib triage
+%if "%{_lib}" == "lib"
+  %global cpuarch 32
+%else
+  %global cpuarch 64
```
Comment (mfrw) [inline SPECS-EXTENDED/gdal/gdal.spec:24]: We only support `64` bit

File: SPECS-EXTENDED/gdal/gdal.spec [hunk 2]
```diff
@@ -0,0 +1,1255 @@
...
+%endif
+%global with_mysql 0
+%global mysql --without-mysql
+%global with_poppler 0
+%global poppler --without-poppler
+%global with_spatialite 0
+%global spatialite --without-spatialite
+%global with_python3 1
+%global python3 --with-python3
+%global with_java 0
```
Comment (mfrw) [inline SPECS-EXTENDED/gdal/gdal.spec:34]: Lets move these macros to the top

File: SPECS-EXTENDED/gdal/gdal.spec [hunk 3]
```diff
@@ -0,0 +1,1255 @@
...
+# Multilib
+# - cpl_config.h is arch-dependent (contains various SIZEOF defines)
+# - gdal-config stores arch-specific information
+mv %{buildroot}%{_includedir}/%{name}/cpl_config.h %{buildroot}%{_includedir}/%{name}/cpl_config-%{cpuarch}.h
+cp -a %{SOURCE2} %{buildroot}%{_includedir}/%{name}/cpl_config.h
+mv %{buildroot}%{_bindir}/%{name}-config %{buildroot}%{_bindir}/%{name}-config-%{cpuarch}
+cp -a %{SOURCE3} %{buildroot}%{_bindir}/%{name}-config
+
+
+%if 0%{run_tests}
```
Comment (mfrw) [inline SPECS-EXTENDED/gdal/gdal.spec:224]: I think we should drop this conditional.
We have `RUN_CHECK=y/n` to run this conditionally.

Also, during tests internet is availible, we might want to install `pytest>=6.0.0` and ensure we do not skip any potential tests.

Comment (arc9693) [reply SPECS-EXTENDED/gdal/gdal.spec]: I ran with pytest, some of the auto (python) based tests themself are not correct, which is why I have ignored them as mentioned in the changelog(updated).

**Review by mfrw — APPROVED** [review]
LGTM

**Pull request merged**